app/code/Magento/Customer/Block/CustomerScopeData.php

```diff
@@ -50,4 +54,37 @@ public function getWebsiteId()
     {
         return (int)$this->_storeManager->getStore()->getWebsiteId();
     }
+
+    /**
+     * @return array
+     */
+    public function getInvalidationRules()
+    {
+        return [
+            '*' => [
+                'Magento_Customer/js/invalidation-processor' => [
+                    'invalidationRules' => [
+                        'website-rule' => [
+                            'Magento_Customer/js/invalidation-rules/website-rule' => [
+                                'scopeConfig' => [
+                                    'websiteId' => $this->getWebsiteId(),
+                                ]
+                            ]
+                        ]
+                    ]
+                ]
+            ],
+        ];
+    }
+
+    /**
+     * Get the invalidation rules json encoded
+     *
+     * @return bool|string
+     * @throws \InvalidArgumentException
+     */
+    public function getSerializedInvalidationRules()
+    {
+        return $this->serializer->serialize($this->getInvalidationRules());
```

Comment:
Still... What are the advantages compared to `echo json_encode($block->getInvalidationRules();` in template? As to me encoding is purely responsibility of template.

Another thing which just came to my mind: do we have a recommendation to leave JS component declaration in templates? Thus they may be overridden more easily, without touching PHP classes. At least I was not able to find similar occurrence of `/js/` or `'*'` in `*.php`.